src/components/features/charactersAndCampaigns/RollDisplay/NormalRollActions.tsx

```diff
@@ -196,6 +207,20 @@ export function NormalRollActions(props: NormalRollActionsProps) {
                 <ListItemText>Burn Momentum</ListItemText>
               </MenuItem>
             )}
+          {(!campaignId || isGM && !characterId) && (
+            <MenuItem
+              onClick={(evt) => {
+                evt.stopPropagation();
+                setIsMenuOpen(false);
+                removeLog(rollId);
```

Comment:
Q: When deleting things around the app I often ask for confirmation before deleting - do you think that rolls are important enough to need that confirmation dialog, or would that just get annoying when trying to mass delete things?

Reply:
Hmm.. that's a good question. I also thought about it. I guess for clearing the whole log (step 2 on implementation of delete Rolls. hehe) a confirmation should be needed.
But here, I'm not sure. It would probably be annoying.

An ideal solution: I don't know how easy it is to clone Rolls state, but displaying a virtualization of the real state. Then when the user deletes a Roll, we flag it as deleted, with an option of "undo" or "put it back". And if in 30 seconds it is not put back, the action is really triggered against Firebase.

![image](https://github.com/scottbenton/Iron-Fellowship_and_Crew-Link/assets/39792862/a9b9c639-620e-4817-9b10-d15b52bf41a6)

What do you think? Is it reasonable?

Reply:
That is a great solution. Definitely better than getting the dialog each time! If you end up looking for an easier option (from an implementation perspective) - you could just have a snackbar pop up with an undo button. They would have until the configurable auto dismiss interval to undo it.

Whichever way you choose, the database change should be fairly easy. Instead of deleting documents we could then just add a hidden: true field to the game log, and update our snapshot listener to filter those out. If I ever feel the need to later clean those up, creating a cleanup job that runs once a day and removes those shouldn't be too bad!